components/CredentialSwitch.vue

```diff
@@ -37,49 +39,47 @@ const props = defineProps({
 
 const selection = computed(() => {
   const {credentialDisplay: {registration: registrations}} = config;
-  const map = {};
+  const map = new Map();
   for(const registration of registrations) {
     const {acceptableTypes, components} = registration;
-    //loop through types so that map will consist of an object
-    //containing all registered types
+    // loop through types so that map will consist of an object
+    // containing all registered types
     for(const type of acceptableTypes) {
-      //If the type already exists in the map, merge it with the
-      //new information
-      if(map.hasOwnProperty(type)) {
-        const existing = map[type];
-        const combined = merge(existing, components);
-        map[type] = combined;
+      // if the type already exists in the map, merge it with the
+      // new information
+      if(map.has(type)) {
+        const combined = extend({
+          target: {}, source: [map.get(type), components], deep: true
```

Comment:
The present implementation hits the map twice, instead hit it just once:

```suggestion
      const existing = map.get(type);
      if(existing) {
        const combined = extend({
          target: {}, source: [existing, components], deep: true
```